.github/workflows/lint_and_test.yaml

```diff
@@ -0,0 +1,31 @@
+name: Lint and Test
```

Comment:
Why lint and test? when separate, they give two different information types, and when together it will first fail on lint, then test and not communicate

Reply:
I did not get you? do you not want check lint and test in same run?

Reply:
i prefer checking them in separate runs because i would like to know the status of both.
- if test fails, i really need to fix something
- if lint fails, it can wait for the next commit. i should fix it but less rush
so having the two workflows gives me two status indicators

Reply:
done